src/kudu/consensus/raft_consensus.h

```diff
@@ -1011,6 +1011,8 @@ class ConsensusRoundHandler {
 
   virtual Status StartFollowerTransaction(const scoped_refptr<ConsensusRound>& context) = 0;
 
+  virtual Status StartConsensusOnlyRound(const scoped_refptr<ConsensusRound>& context) = 0;
```

Comment:
Maybe rename this function and call it something else? It gets called from `StartConsensusOnlyRoundUnlocked` but the name makes me assume that it would take locks and then call `StartConsensusOnlyRoundUnlocked`

Reply:
ah... I see your point. I can see why it is confusing.

If you look at StartFollowerTransactionUnlocked, that uses the same naming convention. It calls StartFollowerTransaction from StartFollowerTransactionUnlocked. The StartConsensusOnlyRound and StartFollowerTransaction confusing calls are members of RoundHandler and not the RaftConsensus:: class, and that class does not adhere to the locking naming conventions of its caller.